interfaces.go

```diff
@@ -44,14 +44,19 @@ type DirectedGraph[NodeType any] interface {
 	Clone() DirectedGraph[NodeType]
 	// HasCycles performs cycle detection and returns true if the DirectedGraph has cycles.
 	HasCycles() bool
-	// PopReadyNodes returns of a list of all nodes that have finalized their status, whether
-	// resolved or unresolvable, and clears the list.
+	// PopReadyNodes returns of a list of all nodes that have no outstanding required dependencies,
+	// and clears the list.
 	// A node becomes ready to process when all of its AND dependencies and at least one of
 	// its OR dependencies are resolved.
+	// Note that nodes can be in any resolution state, since self-resolution is independent of its
+	// own dependency status. Some scenarios that can result in non-waiting nodes being included
+	// in the output include resolving them asynchronously by retrieving nodes independently of
+	// PopReadyNodes(), or due to a propagation of Unresolvable status to dependent nodes.
 	PopReadyNodes() map[string]ResolutionStatus
 	// HasReadyNodes checks to see if there are any ready nodes without clearing them.
 	HasReadyNodes() bool
 	// PushStartingNodes searches for the initial ready nodes without dependencies and saves them.
+	// This includes nodes that have already been marked resolved or unresolvable.
 	// The nodes can then be retrieved with a call to `PopReadyNodes()`.
 	// Recommended to be called only once following construction of the DAG.
```

Comment:
I think that this might be a case of "less is more":  what this function really does is populate the list to be returned by `PopReadyNodes()`, and the less that we say about it probably the better.  That is, if we can, let's let the description of `PopReadyNodes()` do the heavy lifting, and just refer to it (implicitly) here.  So, we probably don't need much more than,
```go
	// PushStartingNodes initializes the list which is retrieved from `PopReadyNodes()`.
	// Recommended to be called only once following construction of the DAG.
```
If someone wants to know how that list is formulated, they can go look at the description on `PopReadyNodes()`, and we don't have to argue about what should or should not be said here.

Reply:
Okay. I guess that works. Done, with one edit.